alphagov/paas-product-pages
# Reimplementation of Google analytics with cookie consent preferences management

## What

Reimplementation of Google analytics to gather insights into usage of our products with cookie consent preferences management.

We inform users about the analytics cookies to measure performance/ usage and ask them if they'd like to opt in.
If they do, we save two GA cookies and the preference setting cookies and load GA script.

Users can opt-out at any point on the /cookies page. 

## How to review

- check out branch
- run `npm install`
- set `this.cookieDomain` to empty string for localhost test in [cookie-functions.js](https://github.com/alphagov/paas-product-pages/pull/99/files#diff-47eef1764be8d04426d2d3c9c5a2f6f54e87c43f43c9b468ee79c2c5b9fe3793R4)
- run `npm run local`
- visit http://localhost:3000
- if you haven't set a preference you will see a banner
- save a cookie preference
- check if have cookies stored (chrome: Option + ⌘ + J (on macOS), or Shift + CTRL + J (on Windows/Linux) - Application tab

or **[view it here](https://sleepy-bardeen-5477f7.netlify.app/)**


![Screenshot 2020-12-11 at 08 48 15](https://user-images.githubusercontent.com/3758555/101882327-a59eb000-3b8d-11eb-9166-8a564917dbc5.png)

![Screenshot 2020-12-11 at 08 48 50](https://user-images.githubusercontent.com/3758555/101882370-b64f2600-3b8d-11eb-83bc-43439492c624.png)

![Screenshot 2020-12-11 at 08 49 11](https://user-images.githubusercontent.com/3758555/101882409-c2d37e80-3b8d-11eb-8f5d-45cf17b163d5.png)
![Screenshot 2020-12-11 at 08 49 42](https://user-images.githubusercontent.com/3758555/101882469-d848a880-3b8d-11eb-9763-ae301a1245ba.png)
